datafusion/proto/src/logical_plan/from_proto.rs

```diff
@@ -698,7 +698,11 @@ impl TryFrom<&protobuf::ScalarValue> for ScalarValue {
                 .map_err(DataFusionError::ArrowError)
                 .map_err(|e| e.context("Decoding ScalarValue::List Value"))?;
                 let arr = record_batch.column(0);
-                Self::List(arr.to_owned())
+                match value {
+                    Value::ListValue(_) => Self::List(arr.to_owned()),
+                    Value::FixedSizeListValue(_) => Self::FixedSizeList(arr.to_owned()),
+                    _ => unreachable!(),
```

Comment:
I think in theory this code could be reached with malformed input -- it might be nice to return an error rather than panic

Reply:
Which panic are you referring to?

If you mean the `unreachable!()`, this statement is inside a match guard that guarantees `value` is one of these two, so I think it is truly unreachable.

Reply:
FYI. For the sake of code robustness, I agree with @alamb 's suggestion to return an error instead of unreachable. If someone accidentally modifies the match branch, this panic might get exposed.